### ISSUE TITLE
[BUG][typescript-fetch] Schema with type object generates undefined function

### ISSUE THREAD

**Issue by rzane** [OpenAPITools/openapi-generator#1877]
##### Description

I'm generating a `typescript-fetch` client using OpenAPI Generator 4.0.0-beta. After running the generator, the following functions are included in the output, but they are not defined:

* `anyFromJSON`
* `anyToJSON`

Also, the type for the property is `any`, but it should be `object` I think?

##### openapi-generator version

4.0.0-beta

##### OpenAPI declaration file content or url

```yaml
openapi: 3.0.0
info:
  title: MyAPI
  description: MY API
  version: 0.0.0
servers:
  - url: http://localhost:8080/api/
paths:
  /example:
    get:
      operationId: getExample
      responses:
        "200":
          description: "Success"
          content:
            application/json:
              schema:
                $ref: "#/components/schemas/Example"

components:
  schemas:
    Example:
      required:
        - data
      properties:
        data:
          type: object
```

##### Command line used for generation

    $ node_modules/.bin/openapi-generator generate -i openapi.yaml -o src -g typescript-fetch

##### Steps to reproduce

I've created a gist that should help to reproduce the issue.

    $ git clone https://gist.github.com/rzane/724ad1ed08721d04966f121461c0737d bug-report
    $ cd bug-report
    $ npm install
    $ npm run generate
    $ npm run test

You should see the following output:

```
> tsc -p src --noEmit

src/models/Example.ts:31:17 - error TS2304: Cannot find name 'anyFromJSON'.

31         'data': anyFromJSON(json['data']),
                   ~~~~~~~~~~~

src/models/Example.ts:40:17 - error TS2304: Cannot find name 'anyToJSON'.

40         'data': anyToJSON(value.data),
                   ~~~~~~~~~
```

The source of the error can be found in `src/models/Example.ts`, which contains the following:

```ts
export interface Example {
    /**
     * 
     * @type {any}
     * @memberof Example
     */
    data: any;
}

export function ExampleFromJSON(json: any): Example {
    return {
        'data': anyFromJSON(json['data']),
    };
}

export function ExampleToJSON(value?: Example): any {
    if (value === undefined) {
        return undefined;
    }
    return {
        'data': anyToJSON(value.data),
    };
}
```

##### Related issues/PRs

I couldn't find one 🤷‍♂️ 

##### Suggest a fix

Ideally, the property `data` would be declared as `object` or empty interface `ExampleItems`, rather than `any`.

More importantly, if the schema is an object without any `properties`  or `additionalProperties`, there should be no coercion.

**Comment by keesvanlierop** [comment]
No updates on this one? I think in the mustache template there should be a condition to check if the datatype that is interpolated is _not_ of type any and just render the value for the field.

**Comment by rzane** [comment]
Definitely agree, though it's not so simple, because you can't do `if value == 'any'` in a mustache template, which is really inconvenient.

From what I can tell, it seems like the `typescript-fetch` client was changed to try to serialize and deserialize objects for the sole purpose of parsing and serializing dates. It's nice that it supports dates, but it also doubles the bundle size of the generated code and increases the complexity immensely. The changes broke a ton of the functionality that was working fine. In this case, I really don't think the benefit of date support was worth all of the negatives that come with the change.

**Comment by wing328** [comment]
I'll take a look tomorrow. Likely it's an issue with the type mapping.

From what I understand, the correct type should be `object`. Please correct me if I'm wrong.

(Disclaimer: no expert in TS)

**Comment by wing328** [comment]
I've filed #2453 to fix the issue and got the following Example.ts
```
// tslint:disable
/**
 * MyAPI
 * MY API
 *
 * OpenAPI spec version: 0.0.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists } from '../runtime';
/**
 *
 * @export
 * @interface Example
 */
export interface Example {
    /**
     *
     * @type {object}
     * @memberof Example
     */
    data: object;
}

export function ExampleFromJSON(json: any): Example {
    return {
        'data': json['data'],
    };
}

export function ExampleToJSON(value?: Example): any {
    if (value === undefined) {
        return undefined;
    }
    return {
        'data': value.data,
    };
}
```

**Comment by yskkin** [comment]
This is not fixed if request is application/x-www-form-urlencoded.

```
    post:
      operationId: create-foo
      responses:
        '200':
          description: OK
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/Foo'
      requestBody:
        content:
          application/x-www-form-urlencoded:
            schema:
              $ref: '#/components/schemas/Foo
```

**Comment by Londeren** [comment]
This is not fixed if request is `multipart/form-data` either:

```
"requestBody": {
    "required": true,
    "content": {
        "multipart\/form-data": {
            "schema": {
                "properties": {
                    "file": {
                        "description": "File content",
                        "type": "file"
                    }
                },
                "type": "object"
            }
        }
    }
},
```

**Comment by nitedani** [comment]
Broken with multipart/form-data

Schema:
```json
properties": {
          "file": {
            "type": "string",
            "format": "binary",
            "title": "File"
          },
          "case_id": {
            "type": "string",
            "title": "Case Id"
          }
}
```

Output:
```js
        if (requestParameters.file !== undefined) {
            formParams.append('file', new Blob([JSON.stringify(anyToJSON(requestParameters.file))], { type: "application/json", }));
                    }

        if (requestParameters.caseId !== undefined) {
            formParams.append('case_id', new Blob([JSON.stringify(anyToJSON(requestParameters.caseId))], { type: "application/json", }));
                    }

**Comment by mstojanovic8** [comment]
Hi, bumping this as it appears to have regressed in recent versions.

We are using **openapi-generator-cli v7.23.0** with the `typescript-fetch` generator and hitting the exact same issue — `anyToJSON is not defined` at runtime for endpoints that have `any`-typed properties (e.g. `photo`, `image`, `document` fields sent as form data blobs).

The generated code references `anyToJSON(...)` directly:
```ts
formParams.append('photo', new Blob([JSON.stringify(anyToJSON(requestParameters['photo']))], { type: "application/json" }));
```

But `anyToJSON` is never defined in `runtime.ts`, causing a `ReferenceError` at runtime.

This was supposedly fixed in PR #2453 back in 2019, but it appears to be broken again in 7.23.0. Could someone confirm whether this is a known regression and if there's a tracked fix or workaround on the roadmap?

As a workaround we are manually adding `anyToJSON` to `runtime.ts` and protecting it from being overwritten via `.openapi-generator-ignore`, but this is not sustainable for an automated SDK generation pipeline.

Is there a version where this is fixed, or is a fix planned?
